plugins/linux-capture/pipewire.c

```diff
@@ -99,6 +106,9 @@ struct _obs_pipewire_data {
 	enum obs_pw_capture_type capture_type;
 	struct obs_video_info video_info;
 	bool negotiated;
+
+	int n_formats;
+	struct modifier_info *modifier_info;
```

Comment:
libobs has [dynamic arrays](https://obsproject.com/docs/reference-libobs-util-darray.html), you can use it here and make the code more native :slightly_smiling_face:

Reply:
Like this, or both format_info and modifiers as darray?

Reply:
Created a commit for both... somewhere the nested darray creates one memory leak per format.